src/containers/setContacts/index.tsx

```diff
@@ -0,0 +1,202 @@
+import { Alert, Spin } from 'antd';
+import moment from 'moment';
+import React, { useEffect } from 'react';
+import { connect, useDispatch } from 'react-redux';
+import { useHistory, useLocation } from 'react-router';
+import protectedApiClient from '../../api/protectedApiClient';
+import { Routes } from '../../App';
+import ContactsForm from '../../components/ContactsForm';
+import { C4CState } from '../../store';
+import {
+  asyncRequestIsComplete,
+  asyncRequestIsFailed,
+} from '../../utils/asyncRequest';
+import { encodeProfileFieldFile } from '../../utils/fileEncoding';
+import {
+  participatingFamilySearchQuery,
+  participatingFamilySearchQueryFlag,
+} from '../../utils/signupFlow';
+import {
+  getContactInfo,
+  resetSetContactState,
+  setContactInfo,
+} from './ducks/thunks';
+import {
+  AdditionalContact,
+  Child,
+  ContactFormFields,
+  ContactInfo,
+  ContactsReducerState,
+} from './ducks/types';
+
+interface SetContactsProps {
+  readonly contacts: ContactsReducerState['contacts'];
+  readonly setContacts: ContactsReducerState['setContacts'];
+}
+
+const SetContacts: React.FC<SetContactsProps> = ({ contacts, setContacts }) => {
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const location = useLocation();
+  useEffect(() => {
+    dispatch(getContactInfo());
+  }, [dispatch]);
+  const registeringAsParticipatingFamily =
+    new URLSearchParams(location.search).get(
+      participatingFamilySearchQueryFlag,
+    ) != null;
+
+  if (asyncRequestIsComplete(setContacts) && registeringAsParticipatingFamily) {
+    protectedApiClient.makePFRequest();
+    history.push({
+      pathname: Routes.SIGNUP_CONFIRMATION,
+      search: participatingFamilySearchQuery,
+    });
+  }
+
+  function selectProfilePictureOrDefaultToUndefined(
```

Comment:
Does this need to be moved to utils? It would only ever be used right here. Needs jsdoc to explain what it does? Better naming?

Reply:
maybe a jsdoc :)